### PR TITLE
build: create script that simplifies bazel testing workflow

### PR DESCRIPTION
**Pull request by devversion**
Instead of remembering what individual browser targets are called,
and what the convention for Bazel labels is, we should just provide
a script that simplifies the Bazel testing workflow. e.g.

```bash
yarn test button [--local] [--firefox] [--no-watch]

yarn test src/material/button src/cdk/bidi
yarn test button slider coercion
```

Eventually we should document this script better in one of the
contributing markdown files, but these docs need a general cleanup
since they still mention Gulp.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,8 +18,9 @@
     "bazel:buildifier": "find . -type f \\( -name \"*.bzl\" -or -name WORKSPACE -or -name BUILD -or -name BUILD.bazel \\) ! -path \"*/node_modules/*\" | xargs buildifier -v --warnings=attr-cfg,attr-license,attr-non-empty,attr-output-default,attr-single-file,constant-glob,ctx-args,depset-iteration,depset-union,dict-concatenation,duplicated-name,filetype,git-repository,http-archive,integer-division,load,load-on-top,native-build,native-package,output-group,package-name,package-on-top,redefined-variable,repository-name,same-origin-load,string-iteration,unused-variable,unsorted-dict-items,out-of-order-load",
     "bazel:format-lint": "yarn -s bazel:buildifier --lint=warn --mode=check",
     "dev-app": "ibazel run //src/dev-app:devserver",
-    "test": "bazel test //src/... --test_tag_filters=-e2e,-browser:firefox-local --build_tag_filters=-browser:firefox-local --build_tests_only",
-    "test-firefox": "bazel test //src/... --test_tag_filters=-e2e,-browser:chromium-local --build_tag_filters=-browser:chromium-local --build_tests_only",
+    "test": "node ./scripts/run-component-tests.js",
+    "test-local": "yarn -s test --local",
```
Comment (crisbeto) [inline package.json]: I'm not sure what `local` refers to in this context. Is it just to indicate that we shouldn't be running this script on the CI?

File: scripts/run-component-tests.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  // If the name does not contain an explicit package name, we try guessing the
+  // package name by walking through an ordered list of possible packages and checking
+  // if a package contains a component with the given name. The first match will be used.
+  for (let guessPackage of orderedGuessPackages) {
+    const guessTargetName = convertPathToBazelLabel(path.join(packagesDir, guessPackage, name));
+    if (guessTargetName !== null) {
+      return guessTargetName;
+    }
+  }
+  throw Error(`Could not find Bazel test target for specified component: ${name}`);
```
Comment (crisbeto) [inline scripts/run-component-tests.js]: Maybe we should print out which names it tried to guess against?

File: package.json (hunk 1 above)
Comment (devversion) [reply package.json]: We decided in another PR that we want to align the old `gulp:static` stuff with "local" that aligns with what is used internally in Google. I didn't feel strongly about it.

File: scripts/run-component-tests.js (hunk 2 above)
Comment (devversion) [reply scripts/run-component-tests.js]: Sure. sounds reasonable

Comment (devversion) [reply scripts/run-component-tests.js]: done.

File: scripts/run-component-tests.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ * using Bazel. Here are a few examples:
+ *
+ *   node ./scripts/run-component-tests button            | Runs Material button tests
+ *   node ./scripts/run-component-tests overlay           | Runs CDK overlay tests
+ *   node ./scripts/run-component-tests src/cdk/a11y      | Runs CDK a11y tests
+ *   node ./scripts/run-component-tests a11y overlay      | Runs CDK a11y and overlay tests
+ *
+ * Supported command line flags:
+ *
+ *   --local    | If specified, no browser will be launched.
```
Comment (jelbourn) [inline scripts/run-component-tests.js]: Can we make this the default? Then invert the flag to something like `--no-headless`

Comment (devversion) [reply scripts/run-component-tests.js]: Do you mean that we do not run any browser _by default_? I don't think we have a way to launch Chrome through Bazel without headless (it seems to depend on the `$DISPLAY` env variable).

Comment (jelbourn) [reply scripts/run-component-tests.js]: I want the command to not pop open a browser window but to also run the tests without having to manually open the page (but you can still open the debug page if you need to)

Comment (devversion) [reply scripts/run-component-tests.js]: I don't think this is possible. The `@bazel/karma` rules decide based on the platform environment whether the browser runs in headless or non-headless mode. For me on Windows, it always runs headless.

**Comment by devversion** [comment]
@Splaktar Yeah. I have an item on my list to update it to match our latest changes. IIRC it still references Gulp etc. If you want to have a look at the doc, that would be cool too 😄

**Comment by devversion** [comment]
@jelbourn The PR that supposedly fixed the `_local` targets landed in `master`. I think this is ready for another round of review. Note that the `_local` target doesn't work out nicely on Windows though (due to a Bazel 1.0.0 change in behavior). Greg reported an issue on the Bazel repo for this. There is also an easy workaround on windows.

**Pull request merged**